### PR TITLE
Bypass MariaDB performance bug of the "IN" sub-query, fix incorrect IssueIndex

### LINKED ISSUE #26277: Use join for the deleting issue actions query
### Description

Our current DELETE query for actions is rather inefficient on MariaDB:

```sql
DELETE FROM action WHERE comment_id IN (SELECT id FROM comment WHERE issue_id=?)
```

https://github.com/go-gitea/gitea/blob/ab388deb0e52c058a19dbd844bdd890f7cf84d51/models/activities/action.go#L688-L702

Caused by https://jira.mariadb.org/browse/MDEV-16289

see https://codeberg.org/forgejo/forgejo/issues/1161 and https://codeberg.org/forgejo/forgejo/pulls/1165

### Gitea Version

1.20.2

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Binary started via systemd behind a caddy reverse proxy.

### Database

None

### PR DESCRIPTION
**Pull request by wxiaoguang**
Close #26277
Fix #26285

### CODE REVIEW

**Comment by wxiaoguang** [comment]
> BTW we don't actually test against MariaDB, we should probably add it to the db tests.

By this approach, no need to test it with any particular database, because it only uses simple "SELECT" and simple "DELETE", they are standard SQL.

**Comment by wxiaoguang** [comment]
If you mean that we need to add a MariaDB for all test, I think it's not a must at the moment.

Although MariaDB is a hard-fork of MySQL, actually, they share the same syntax for MySQL 5.7 and the syntax is quite similar as MySQL 8.

As long as Gitea doesn't use any minority MySQL 8 feature, the current MySQL test is good enough.

**Comment by silverwind** [comment]
Not really an expert in this topic, but I consider those two forks to have diverged enough by this point that they can not be considered the same database anymore. Of course if you limit yourself to simple standard SQL (and I hope xorm does), they should be compatible.

**Comment by wxiaoguang** [comment]
I am heavily using MySQL and MariaDB in my projects, AFAIK they are totally different databases now but the majority statements (syntax) are the same.

Maybe we can drop MySQL 5.7 support and involve MariaDB 10.x test in next release.

ps: XORM has already had some problems for various database versions (even for MySQL) nowadays, for example, the incorrect startup warnings, etc.

File: models/activities/action.go [hunk 1]
```diff
@@ -685,18 +685,34 @@ func NotifyWatchersActions(acts []*Action) error {
...
+		} else {
+			lastCommentID = commentIDs[len(commentIDs)-1]
+		}
 	}
 
-	_, err := db.GetEngine(ctx).Table("action").Where("repo_id = ?", repoID).
+	_, err := e.Where("repo_id = ?", repoID).
 		In("op_type", ActionCreateIssue, ActionCreatePullRequest).
-		Where("content LIKE ?", strconv.FormatInt(issueID, 10)+"|%").
+		Where("content LIKE ?", strconv.FormatInt(issueIndex, 10)+"|%"). // "IssueIndex|content..."
```
Comment (lunny) [inline models/activities/action.go:715]: Is this a bugfix? Before it's issueID and after it's issueIndex?

Comment (wxiaoguang) [reply models/activities/action.go:715]: Yes, it is a bug, reported by #26285

**Review by lunny — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Pull request merged**